driver-core/src/main/java/com/datastax/driver/core/ErrorTracker.java

```diff
@@ -0,0 +1,50 @@
+/*
+ *      Copyright (C) 2012-2015 DataStax Inc.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+package com.datastax.driver.core;
+
+/**
+ * Interface for objects that are interested in monitoring precisely errors returned by all hosts.
+ *
+ * An implementation of this interface can be registered against a Cluster
+ * object trough the {@link Cluster#register} method, after which the
+ * {@link #notifyError(Host, Statement, Exception)} method will be called after each query of the driver to a Cassandra
+ * host with the error that has been returned for this operation.
+ */
+public interface ErrorTracker {
+
+    /**
+     * Method that is called after each request returning an error code from Cassandra.
+     *
+     * @param host the host that has been queried.
+     * @param statement the statement that has been sent.
+     * @param exception the exception the query caused.
+     */
+    public void notifyError(Host host, Statement statement, Exception exception);
```

Comment:
I'm not a big fan of `ErrorTracker` neither. It looks very close to `LatencyTracker` – I even tried locally to get rid of it and replace it with `LatencyTracker`, it seems very easy to do.
OK, I understand that the tracker would be called much more frequently if it were a `LatencyTracker`, but I'm not sold either on the usefulness of a new tracker just to avoid that. Besides, some policies could mix both error-awareness and latency-awareness to compute the score of a host, so having one same tracker for both errors and latencies seems easier.
Now, the name `LatencyTracker` is probably a bad one in that it probably doesn't reflect all the possible usages of such trackers, since it stress the latency aspect of them, but not the others (e.g. they also get notified of every executed statement, and every error, and these aspects are also very useful in different ways: cf. `QueryLogger`, etc).